middlewares/authentication.go

```diff
@@ -0,0 +1,19 @@
+package middlewares
```

Comment:
In terms of file name, wouldn't it be better to name the file closer to what the middleware name is `ensure_no_authentication.go`? That allows to a faster code discovery. If you don't having the `no` in the middle, then `ensure_guest_user.go` might sound better.

Reply:
In the first place, I thought to put all the authentication related middleware in the `authentication.go` such as `ensure_no_authentication`, `ensure_authentication` and so on.

But I agree with you if we named the file closer to the middleware name allows a faster code discovery so I fixed in f0ef5e0023c462ed2188d8b13e40dcb04f5419a3